### DISCUSSION TITLE
Interest in Azure Functions packages

### DISCUSSION THREAD

**Discussion by tpwalke2** [graphql-dotnet/server#965] [Ideas]
I was recently working on a GraphQL proof of concept and chose to host the endpoint in an Azure Function HTTP trigger. The endpoint was fairly straightforward and I was able to demonstrate the custom querying ability. My main concern was how to create documentation that automatically evolves along with the schema (similar to Swagger). It's clear that GraphiQL, et. al. are contenders for that living documentation.

I saw that the server.UI packages provide these tools, but noticed that they are built distinctly for the ASP.NET Core request pipeline. For the proof of concept, I made an adapter to sit between the middleware and an HTTP trigger and got it running. However, this is a fragile arrangement and unsuited for deployment to production workloads.

Is there interest in making Azure function aware analogs for the server.UI packages? If so, should they be separate NuGet packages altogether? Or should they sit in the same existing packages? My initial thought would be to make separate packages to reduce unnecessary dependencies.

I wouldn't mind working on them, but would need assistance with figuring out NuGet packaging.

**Comment by sungam3r** [comment]
For a long time we have some thoughts to move UI packages out of this repo. There are pros and cons. Yes, UI logic does not use server code at all. You may start your work here, trying to achieve your goals. Then we can evaluate either keep this new code here or move out.

**Comment by Shane32** [comment]
I'd suggest simply adding an `ActionResult` to the UI packages.  I wrote an example here:

https://github.com/graphql-dotnet/server/blob/azure_functions/src/Ui.Playground/PlaygroundActionResult.cs

This requires an additional dependency for .NET Standard 2.0 targets to compile properly, but not .NET 6 targets.

https://github.com/graphql-dotnet/server/pull/966/files#diff-05512c84fe1bef66910d404bce216b922927e69c1e54a77ed6405fc3b801a428

I also worked on adding Azure Functions support into the server project.  The same PR has a new sample Azure Function sample application using the same Chat schema as the rest of the samples.  It doesn't seem that Azure Functions support WebSockets, so no subscription support, but otherwise it seems to work perfectly.

See:
- https://github.com/graphql-dotnet/server/pull/966

Please let me know your thoughts.  To finish the PR, I would need to:
- Add similar ActionResult classes to the other UI projects
- Add documentation in the readme for usage within Azure Functions
- Add tests to ensure that the new sample application works correctly
